csm/plugins/cortx/rgw.py

```diff
@@ -0,0 +1,105 @@
+# CORTX-CSM: CORTX Management web and CLI interface.
+# Copyright (c) 2022 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+from typing import Any
+from json import loads
+from csm.core.services.rgw.s3.utils import CsmRgwConfigurationFactory
+from csm.core.data.models.rgw import RgwErrors, RgwError
+from csm.common.errors import CsmInternalError
+from cortx.utils.log import Log
+from csm.core.blogic import const
+from cortx.utils.rgwadmin import RGWAdminClient
+from cortx.utils.rgwadmin import RGWAdminClientException
+
+
+class RGWPlugin:
+
+    def __init__(self):
+        """
+        Initialize RGW plugin
+        """
+        config = CsmRgwConfigurationFactory.get_rgw_connection_config()
+        self._rgw_admin_client = RGWAdminClient(config.auth_user_access_key,
+            config.auth_user_secret_key, config.host, config.port)
+        Log.info(f"RGW admin uid: {const.ADMIN_UID}")
+        self._api_operations = {
```

Comment:
Minor: could it be placed into a template file?

Reply:
@shrimetta could you please look into this ?

Reply:
@avrnv completed